gatsby-node.js

```diff
@@ -40,3 +44,65 @@ exports.onCreateWebpackConfig = ({ stage, actions, getConfig }) => {
     actions.replaceWebpackConfig(config)
   }
 }
+
+const checkIfDirExists = async folderPath => {
+  try {
+    await stat(folderPath)
+    return true
+  } catch (err) {
+    if (err.code === 'ENOENT') {
+      return false
+    }
+  }
+}
+
+const mkDirIfItDoesNotExist = async folderPath => {
+  const doesDirExist = await checkIfDirExists(folderPath)
+  if (!doesDirExist) {
+    await mkdir(folderPath, { recursive: true })
+  }
+}
+
+exports.onPostBuild = async ({ graphql }) => {
+  try {
+    // const start = Date.now()
+    const {
+      data: {
+        allBlogPost: { nodes }
+      }
+    } = await graphql(`
+      query GetBlogMainImages {
+        allBlogPost {
+          nodes {
+            picture {
+              fields {
+                sourcePath
+              }
+            }
+            date
+          }
+        }
+      }
+    `)
+
+    await Promise.all(
+      nodes.map(async node => {
+        const imagePath = node.picture.fields.sourcePath
+        const dirPath = path.dirname(
+          path.join(__dirname, 'public', 'blog', imagePath)
+        )
+        await mkDirIfItDoesNotExist(dirPath)
+        return sharp(path.join(__dirname, 'static', 'uploads', imagePath))
+          .resize({ width: 850 })
+          .toFile(path.join(__dirname, 'public', 'blog', imagePath))
+          .catch(err => {
+            console.error(err)
+          })
+      })
+    )
+    // const end = Date.now()
```

Comment:
please, let's not commit commented or dead code

Reply:
Comments deleted!